packages/cli/internal/pkg/cli/account_activate.go

```diff
@@ -81,7 +85,20 @@ func (o *accountActivateOpts) Execute() error {
 		environmentVars = append(environmentVars, fmt.Sprintf("VPC_ID=%s", o.vpcId))
 	}
 
-	return o.deployCoreInfrastructure(environmentVars)
+	homeDir, err := osutils.DetermineHomeDir()
+	if err != nil {
+		return err
+	}
+
+	cdkAppPath := filepath.Join(homeDir, cdkCoreDir)
+	if !o.skipBootstrap {
```

Comment:
Why are we forcing the default behavior to be deploy the bootstrap? 
Could we not check to see if the bootstrapped CDK stack exists in an active state and if it doesn't then try and deploy it? We could also use a flag to force a bootstrap deploy if the customer requested it.

Reply:
'cdk bootstrap' is effectively a noop on deployed bootstrap stacks, with the caveat that it might update an older bootstrap template (which would be backwards compatible), there isn't a good way of checking if the bootstrap stack exists without hardcoding something that would likely break someday anyway since there are no guarantees on things like bootstrap stack name.

Reply:
Won't this effective stomp on the customer's previous bootstrap if they used any customization? e.g. [this stuff](https://docs.aws.amazon.com/cdk/v2/guide/bootstrapping.html#bootstrapping-customizing)

To avoid this I think we could use a customer qualifier/synthesize to effectively have a "custom" bootstrap stack that is only used by AGC?

Reply:
A custom qualifier alone doesn't produce a separate bootstrap stack, so even if we used a custom qualifier it might mess with resource in the default bootstrap stack... However, I did find `--toolkit-stack-name`, which unfortunately is not referenced in the cdk / bootstrapping developer guide. So `cdk bootstrap  --toolkit-stack-name Agc-CDKToolkit --qualifier agc` will produce a separate bootstrap stack (`Agc-CDKToolkit`) with the same bootstrap resources as the default bootstrap stack (the qualifier is used to remove resource naming conflicts with the default bootstrap stack). 

So effectively yeah, we could have an `Agc-CDKToolkit` bootstrap stack that we manage ourselves with the downside that is replicates the resources of the default bootstrap stack, which is additional (albeit tiny) infra + cost  for the user. 

This should alleviate your and @elliot-smith's concerns